### PR TITLE
feat(quantic): improved accessibility of the Quantic Generated Answer component

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-5243](https://coveord.atlassian.net/browse/SFINT-5243)

Accessibility of the Quantic Generated Answer component improved:

1. Feedback buttons are now using the Quantic Stateful Button that comes with the a11y logic that mentions the state of the button to the screen reader.
2. New aria live region is now created by the Quantic Generated Answer to give the status of the generated answer and read the answer.
3. Aria live region announces the following: 

- when the generated answer is loading or streaming.
- when the generated answer is ready, it reads the answer in this situation.
- when the generated answer is hidden.
- when an error occurs.

#### Demo:

https://github.com/coveo/ui-kit/assets/86681870/487c51c1-2bd9-4bb7-9c9a-c69c161f061a

    

[SFINT-5243]: https://coveord.atlassian.net/browse/SFINT-5243?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 1]
```diff
@@ -162,6 +172,28 @@ export default class QuanticGeneratedAnswer extends LightningElement {
   updateState() {
     this.state = this.generatedAnswer.state;
     this.updateFeedbackState();
+    console.log(this.getGeneratedAnswerStatus());
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:175]: console.log? :P

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 2]
```diff
@@ -162,6 +172,28 @@ export default class QuanticGeneratedAnswer extends LightningElement {
...
+      return this.labels.couldNotGenerateAnAnswer;
+    }
+    if (isGenerating) {
+      return this.labels.generatingAnswer;
+    }
+
+    return hasAnswer
+      ? I18nUtils.format(this.labels.answerGenerated, this.answer)
+      : '';
   }
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:197]:
```suggestion
  getGeneratedAnswerStatus() {
    if (!this.state.isVisible) {
      return this.labels.generatedAnswerIshidden;
    }

    if (this.hasRetryableError) {
      return this.labels.couldNotGenerateAnAnswer;
    }
    
    const isGenerating = this.state.isLoading || this.state.isStreaming;
    if (isGenerating) {
      return this.labels.generatingAnswer;
    }
    
    const hasAnswer = !!this.state.answer;
    return hasAnswer
      ? I18nUtils.format(this.labels.answerGenerated, this.answer)
      : '';
  }
```

Small clarity improvement IMO

File: packages/quantic/force-app/main/default/lwc/quanticStatefulButton/quanticStatefulButton.js [hunk 3]
```diff
@@ -43,6 +43,23 @@ export default class QuanticStatefulButton extends LightningElement {
...
+   */
+  @api
+  get size() {
+    return this._size;
+  }
+  set size(value) {
+    if (['xx-small', 'x-small', 'small', 'medium', 'large'].includes(value)) {
+      this._size = value;
+    }
+  }
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticStatefulButton/quanticStatefulButton.js:59]:
```suggestion
  /**
   * The size of the icon.
   * @api
   * @type {'xx-small' | 'x-small' | 'small' | 'medium' | 'large'}
   */
  @api
  get iconSize() {
    return this._size;
  }
  set iconSize(value) {
    if (['xx-small', 'x-small', 'small', 'medium', 'large'].includes(value)) {
      this._size = value;
    }
  }
```

For clarity

File: packages/quantic/force-app/main/translations/fr.translation-meta.xml [hunk 4]
```diff
@@ -752,4 +752,16 @@
     <label>Résumé</label>
     <name>quantic_Summary</name>
   </customLabels>
+  <customLabels>
+    <label>La réponse générée est cachée</label>
```
Comment (erocheleau) [inline packages/quantic/force-app/main/translations/fr.translation-meta.xml:756]: Ca ne serait pas plutôt: "Le générateur de réponse est désactivé" ?

File: packages/quantic/force-app/main/translations/fr.translation-meta.xml [hunk 5]
```diff
@@ -752,4 +752,16 @@
...
+    <name>quantic_GeneratedAnswerIsHidden</name>
+</customLabels>
+<customLabels>
+  <label>Réponse générée: {{0}}</label>
+  <name>quantic_AnswerGenerated</name>
+</customLabels>
+<customLabels>
+  <label>Génération de la réponse</label>
```
Comment (erocheleau) [inline packages/quantic/force-app/main/translations/fr.translation-meta.xml:764]:
```suggestion
  <label>En attente de la réponse</label>
```

Maybe?

**Review by erocheleau — APPROVED** [review]
Two small changes for clarity that I suggest. But other than that it all looks good to me!

File: packages/quantic/force-app/main/translations/fr.translation-meta.xml (hunk 4 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/translations/fr.translation-meta.xml:756]: Could be yeah, but in reality we are not really disabling it, we are just hiding it, that's why "cachée"  seems  to me to be the right one.

File: packages/quantic/force-app/main/translations/fr.translation-meta.xml (hunk 5 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/translations/fr.translation-meta.xml:764]: the screen reader announces this label while the generating answer is streaming, so yes we are waiting for the answer but more precisely the answer is being generated.

**Pull request merged**